kv/kvclient/kvcoord: TestProxyTracing failed

kv/kvclient/kvcoord.TestProxyTracing [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21339594?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21339594?buildTab=artifacts#/) on master @ [c968ad01c4089462ab483d7e850b33f314578c0e](https://github.com/cockroachdb/cockroach/commits/c968ad01c4089462ab483d7e850b33f314578c0e):


```
=== RUN   TestProxyTracing
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/a38679e165d5c98145544b9bfab9e765/logTestProxyTracing1303401316
    test_log_scope.go:82: use -show-logs to present logs inline
    dist_sender_server_test.go:4454: -- test log scope end --
--- FAIL: TestProxyTracing (58.30s)
=== RUN   TestProxyTracing/lease-type=LeaseExpiration
    test_server_shim.go:146: using test tenant configuration from testserver factory defaults
    test_server_shim.go:150: cluster virtualization disabled due to issue: #156327 (expected label: C-bug)
    dist_sender_server_test.go:4426: condition failed to evaluate within 45s: from dist_sender_server_test.go:4403: expected 3 leases on node 3, found 1
        
        goroutine dump: /artifacts/tmp/_tmp/a38679e165d5c98145544b9bfab9e765/goroutine_dump_176027785.txt
--- FAIL: TestProxyTracing/lease-type=LeaseExpiration (50.64s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv-triage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestProxyTracing.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-63679

--- a/pkg/kv/kvclient/kvcoord/dist_sender_server_test.go
+++ b/pkg/kv/kvclient/kvcoord/dist_sender_server_test.go
@@ -4398,8 +4398,10 @@ func TestProxyTracing(t *testing.T) {
 
 		checkLeaseCount := func(node, expectedLeaseCount int) error {
 			if count := leaseCount(node); count != expectedLeaseCount {
-				require.NoError(t, tc.GetFirstStoreFromServer(t, 0).
-					ForceLeaseQueueProcess())
+				for i := 0; i < tc.NumServers(); i++ {
+					require.NoError(t, tc.GetFirstStoreFromServer(t, i).
+						ForceLeaseQueueProcess())
+				}
 				return errors.Errorf("expected %d leases on node %d, found %d",
 					expectedLeaseCount, node, count)
 			}

